designs/2019-linter-options/README.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 2019-05-12
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# The `linterOptions` property in Config Files
+
+## Summary
+
+This RFC adds `linterOptions` property to config files. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
+
+## Motivation
+
+We cannot configure some linter's behavior with config files, especially, shareable configs. It's convenient if we can configure those in shareable configs.
+
+## Detailed Design
+
+This adds `linterOptions` property to config files with three properties.
+
+```js
+// .eslintrc.js
+module.exports = {
+    linterOptions: {
+        allowInlineConfig: true,              // Corresponds to --no-inline-config / `options.allowInlineConfig`
+        reportUnusedDisableDirectives: "off", // Corresponds to --report-unused-disable-directives / `options.reportUnusedDisableDirectives`
+        ignorePatterns: [],                   // Corresponds to --ignore-pattern / `options.ignorePattern`
```

Comment:
If https://github.com/eslint/rfcs/pull/20 is implemented, would `ignorePatterns` effectively be the same as `excludedFiles`? It seems like they fill very similar use cases.

Reply:
No, `excludedFiles` and `ignorePatterns` are different.

- `excludedFiles` - If matched, this override entry is not applied, but ESLint does lint the file (if any of the other override entry, `--ext` or glob patterns was matched).
- `linterOptions.ignorePatterns` - If matched, ESLint does not lint the file.

With #20, indeed, there is some situation that the two is matched. But those are different.

Reply:
In implementation details:

- if `excludedFiles` matched and all of the other override entries, `--ext`, and globs didn't match, then the file enumerator doesn't yield the file entry because it's not a target file.
- if `linterOptions.ignorePatterns` matched, then the file enumerator yields the file entry with `flag: IGNORED_SILENTLY` because it's a target file but ignored.